expression/builtin_string_vec_generated.go

```diff
@@ -0,0 +1,161 @@
+)
+
+// vecEvalInt evals FIELD(str,str1,str2,str3,...).
+// See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_field
+func (b *builtinFieldIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+
+	arg0 := buf0.Int64s()
+
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+	for i := 1; i < len(b.args); i++ {
+		if err := b.args[i].VecEvalInt(b.ctx, input, buf1); err != nil {
+			return err
+		}
+		buf1.MergeNulls(buf0)
+
+		arg1 := buf1.Int64s()
+
+		for j := 0; j < n; j++ {
+			if buf1.IsNull(j) || i64s[j] > 0 {
+				continue
+			}
+
+			if arg0[j] == arg1[j] {
+
+				i64s[j] = int64(i)
+			}
+		}
+	}
+	return nil
+}
+
+func (b *builtinFieldIntSig) vectorized() bool {
+	return true
+}
+
+// vecEvalInt evals FIELD(str,str1,str2,str3,...).
+// See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_field
+func (b *builtinFieldRealSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalReal(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+
+	arg0 := buf0.Float64s()
+
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+	for i := 1; i < len(b.args); i++ {
+		if err := b.args[i].VecEvalReal(b.ctx, input, buf1); err != nil {
+			return err
+		}
+		buf1.MergeNulls(buf0)
+
+		arg1 := buf1.Float64s()
+
+		for j := 0; j < n; j++ {
+			if buf1.IsNull(j) || i64s[j] > 0 {
+				continue
+			}
+
+			if arg0[j] == arg1[j] {
+
+				i64s[j] = int64(i)
+			}
+		}
+	}
+	return nil
+}
+
+func (b *builtinFieldRealSig) vectorized() bool {
+	return true
+}
+
+// vecEvalInt evals FIELD(str,str1,str2,str3,...).
+// See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_field
+func (b *builtinFieldStringSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+	for i := 1; i < len(b.args); i++ {
+		if err := b.args[i].VecEvalString(b.ctx, input, buf1); err != nil {
+			return err
+		}
+		buf1.MergeNulls(buf0)
```

Comment:
Since the type of data in `buf1` is `string` which is a var-length type, it is not recommended to use it as the method receiver of `MergeNulls`.
How about using `buf1.IsNull(j) || result.IsNull(j) || i64s[j] >0`?

Reply:
ok, I forget it